File: src/Systems/Fortran/patterns/monitor_object.f90
module monitor_object_example
  implicit none
  type :: counter
    integer :: value = 0
  contains
    procedure :: add => counter_add
  end type
contains
  subroutine counter_add(self, amount)
    class(counter), intent(inout) :: self
    integer, intent(in) :: amount
    self%value = self%value + amount
  end subroutine
  logical function run()
    type(counter) :: item
    call item%add(2); call item%add(3)
    run = item%value == 5
  end function
end module

File: src/Systems/Swift/patterns/MonitorObject.swift
import Foundation

enum MonitorObjectExample {
    final class Counter {
        private let lock = NSLock()
        private var value = 0

        func add(_ amount: Int) {
            lock.lock()
            defer { lock.unlock() }
            value += amount
        }

        func get() -> Int {
            lock.lock()
            defer { lock.unlock() }
            return value
        }
    }

    static func run() -> Bool {
        let counter = Counter()
        counter.add(2)
        counter.add(3)
        return counter.get() == 5
    }
}
